Patch-Source: https://gitlab.gnome.org/GNOME/gimp/-/commit/9b517fbf7d629fb940344719eddf9421f013569b
--
From 9b517fbf7d629fb940344719eddf9421f013569b Mon Sep 17 00:00:00 2001
From: Stanislav Grinkov
 <43956-stanislavgrinkov@users.noreply.gitlab.gnome.org>
Date: Thu, 9 Feb 2023 23:05:27 +0600
Subject: [PATCH] [app] Ensure calculated histogram spike height ...

is positive or zero. Sometimes the calculated `y` or `bg_y`
values are negative because the histogram value was divided
by zero (when max == 0.0).

Resolves: #9132 (Gnome/GIMP bug tracker)
---
 app/widgets/gimphistogramview.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/app/widgets/gimphistogramview.c b/app/widgets/gimphistogramview.c
index b262440070..4ac3bda895 100644
--- a/app/widgets/gimphistogramview.c
+++ b/app/widgets/gimphistogramview.c
@@ -530,6 +530,9 @@ gimp_histogram_view_draw_spike (GimpHistogramView    *view,
       break;
     }
 
+  y    = MAX (y,    0);
+  bg_y = MAX (bg_y, 0);
+
   if (bg_color)
     {
       gdk_cairo_set_source_rgba (cr, bg_color);
-- 
GitLab

